src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java

```diff
@@ -37,8 +37,12 @@
     extends AbstractViolationReporter
     implements FileSetCheck {
 
-    /** Collects the error messages. */
-    private final SortedSet<LocalizedMessage> messageCollector = new TreeSet<>();
+    /**
+     * Collects the error messages.
+     * @noinspection ThreadLocalNotStaticFinal
+     */
+    private final ThreadLocal<SortedSet<LocalizedMessage>> messageCollector =
```

Comment:
@soon and @sabaka , 
There are some concern with such rule/inspection in web - https://stackoverflow.com/questions/30134272/can-javas-threadlocal-be-applied-to-a-non-static-field-if-yes-how 

please provide a prove why we do not need `static` as it looks like should still work.